archunit/src/test/java/com/tngtech/archunit/library/ArchitecturesTest.java

```diff
@@ -218,6 +219,55 @@ public void layered_architecture_combines_multiple_ignores() {
         assertThat(layeredArchitecture.evaluate(classes).hasViolation()).as("result has violation").isFalse();
     }
 
+    @Test
+    public void layered_architecture_where_all_classes_are_covered_fully() {
+        JavaClasses classes = new ClassFileImporter().importClasses(
+                FirstAnyPkgClass.class,
+                SecondThreeAnyClass.class);
+
+        LayeredArchitecture architecture = layeredArchitecture()
+                .layer("One").definedBy("..first..")
+                .layer("Two").definedBy("..second..")
+                .whereAllClassesAreCovered();
+
+        assertThat(architecture.getDescription()).contains("where all classes are covered");
+        assertThat(architecture.evaluate(classes).hasViolation())
+                .isFalse();
+    }
+
+    @Test
+    public void layered_architecture_where_all_classes_are_covered_with_ignoring() {
+        JavaClasses classes = new ClassFileImporter().importClasses(
+                FirstAnyPkgClass.class,
+                SecondThreeAnyClass.class);
+
+        LayeredArchitecture architecture = layeredArchitecture()
+                .layer("One").definedBy("..first..")
+                .layer("Two").definedBy("..second..")
```

Comment:
Maybe it would be better if the test would fail in case `covered...ignoring...` would not work?

Reply:
Thought I was covering the failure cases too but will check :)

Reply:
Doh. Fixed in 88cbd3e!